Runtime/Scripts/DeveloperConsole.cs

```diff
@@ -0,0 +1,145 @@
+        public abstract string Description { get; protected set; }
+        public abstract string Help { get; protected set; }
+
+        public void AddCommandToConsole()
+        {
+            DeveloperConsole.AddCommandsToConsole(Command, this);
+        }
+
+        public abstract void RunCommand();
+    }
+
+    public class DeveloperConsole : MonoBehaviour
+    {
+        public static DeveloperConsole Instance { get; private set; }
+        public static Dictionary<string, ConsoleCommand> Commands { get; private set; }
+
+        [Header("UI Components")]
+        public Canvas consoleCanvas;
+
+        public Text consoleText;
+        public Text inputText;
+        public InputField consoleInput;
+
+        private void Awake()
+        {
+            if (FindObjectsOfType(GetType()).Length > 1)
+            {
+                Destroy(gameObject);
+            }
+
+            if (Instance == null)
+            {
+                Instance = this;
+            }
+            else if (Instance == this)
+            {
+                Destroy(Instance.gameObject);
+                Instance = this;
+            }
+            DontDestroyOnLoad(this.gameObject);
+            Commands = new Dictionary<string, ConsoleCommand>();
+        }
+
+        private void Start()
+        {
+            consoleCanvas.gameObject.SetActive(false);
+            AddMessageToConsole("Starting Developer Console...");
+            CreateCommands();
+        }
+
+        private void OnEnable()
+        {
+            Application.logMessageReceived += HandleLog;
+        }
+
+        private void OnDisable()
+        {
+            Application.logMessageReceived -= HandleLog;
+        }
+
+        private void HandleLog(string logMessage, string stackTrace, LogType type)
+        {
+            string message = string.Empty;
+            if (type.Equals(LogType.Warning))
+            {
+                message = "<color=yellow>[" + type.ToString() + "]" + logMessage + "</color> ";
+            }
+            else
+            {
+                message = "<color=white>[" + type.ToString() + "]" + logMessage + "</color> ";
+            }
+            AddMessageToConsole(message);
+        }
+
+        private void CreateCommands()
+        {
+            CommandQuit.CreateCommand();
+        }
+
+        public static void AddCommandsToConsole(string name, ConsoleCommand command)
+        {
+            if (!Commands.ContainsKey(name))
+            {
+                Commands.Add(name, command);
+            }
+        }
+
+        private void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.BackQuote))
+            {
+                consoleCanvas.gameObject.SetActive(!consoleCanvas.gameObject.activeInHierarchy);
+            }
+
+            if (consoleCanvas.gameObject.activeInHierarchy)
+            {
+                if (Input.GetKeyDown(KeyCode.Return))
+                {
+                    if (inputText.text != "")
```

Comment:
Could you just have one if statement? 

`if (consoleCanvas.gameObject.activeInHierarchy && Input.GetKeyDown(KeyCode.Return) && inputText.text != "") { ... }`